sim/robots/flood_fill.py

```diff
@@ -561,12 +619,51 @@ def flood_fill_thorough_explorer(  # pylint: disable=too-many-branches,too-many-
         mark_deadends(maze, pos[:-1], start, goals, 'orange')
 
     print(f"flood hunter: found goals! {start=}, {maze.explored_cells_percentage()=:.02%}/{percentage=:.02%}")
+    dead_ends_groups: list[set[tuple[int, int]]] = []
     while maze.explored_cells_percentage() < percentage:
         print(f"flood hunter: pos={tuple(pos)}, {maze.explored_cells_percentage()=:.02%}/{percentage=:.02%}")
         # print(_render_maze(maze, goals=goals, pos=pos))
         mark_unreachable_groups(maze, pos[:-1])
 
-        unknown, all_unknown = _calc_unknown_groups(maze, pos[:-1], start, goals)
+        unknown_color = 'blue'
+        unknown, all_unknown, flood_dead_ends = _calc_unknown_groups(maze, pos[:-1], start, goals, unknown_color=unknown_color)
+        dead_ends_groups += flood_dead_ends
```

Comment:
Add a comment that explains there can be no duplicates here (if there can, this should probably be a `set`)